test/index.js

```diff
@@ -49,6 +49,32 @@ describe('strange-forms', () => {
             done();
         });
 
+        it('does not update a prop in state when it hasn\'t changed.', (done) => {
```

Comment:
This test hits all the new code branches, but doesn't test the new behavior– it also passes without these new changes.  The case we really care about is when `proposeNew(field)(event)` updates the local form state without triggering new props, and then new props come down for some other reason.  Formerly they were overwriting the local component state for fields that actually didn't change.

Reply:
I know it's annoying, haha.  I will try to write a test for it when I find a moment, if that would be useful.